dataflow/gpu-workers/pytorch-minimal/main.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import logging
+from typing import Any, List, Optional
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import PipelineOptions
+import torch
+
+
+def check_gpus(element: Any, gpus_optional: bool = False) -> Any:
+    """Validates that we are detecting GPUs, otherwise raise a RuntimeError."""
+    if torch.cuda.is_available():
+        logging.info(f"Using GPU: {torch.cuda.get_device_name(0)}")
+    elif gpus_optional:
+        logging.warning("No GPUs found, defaulting to CPU.")
+    else:
+        raise RuntimeError("No GPUs found.")
+    return element
+
+
+def run(input_text: str, beam_args: Optional[List[str]] = None) -> None:
+    beam_options = PipelineOptions(beam_args, save_main_session=True)
+
+    # We currently cannot use the `with` statement to run without waiting.
+    #   https://issues.apache.org/jira/browse/BEAM-12455
+    pipeline = beam.Pipeline(options=beam_options)
+    (
+        pipeline
+        | "Create data" >> beam.Create([input_text])
+        | "Check GPU availability" >> beam.Map(check_gpus)
+        | "My transform" >> beam.Map(logging.info)
```

Comment:
We could add a comment that users can sub this with a code that actually uses GPU here or perhaps make it a separate function + add a comment.